file-server/Dockerfile

```diff
@@ -8,12 +8,12 @@ FROM node:14.8.0-alpine
 
 WORKDIR /app
 
-COPY package*.json ./
+COPY file-server/package*.json ./
```

Comment:
It seems a bit counter-intuitive to have the context for a Dockerfile located in the parent directory. Maybe these Dockerfiles can also be moved to the root directory and renamed.
I think there is a convention of being able to run docker commands, e.g. `docker build -t my-file-server .` from within the directory containing the Dockerfile. This change breaks that convention. But maybe it's worth it if it is the best way to solve other problems.
I prefer not to force docker-compose, or a makefile, as the only right way to build and run docker containers. My deployment configuration, for example, does not use docker-compose at all. It builds the docker images directly, using each containing folder as the context for that Dockerfile. I can certainly refactor my deployment stuff, if this does come out as the best way forward, but I would prefer to find a way that does not break the convention of Docker context == folder containing Dockerfile, if that is not too inconvenient.

Reply:
I think I can fix this problem with an `ENV` variable.

Reply:
> I would prefer to find a way that does not break the convention of Docker context

I'll admit feeling the same.

> moves the docker build directories up one level as @patcon suggested earlier

To clarify, I was only intending to share that this is one technical way one would get access to a shared config directory, but I'm not of the position it's the appropriate way for us.

I remain curious whether a simple bash script run in advance (ie. `cp`) would be functionally equivalent to changing. the docker context (which feels like an infra code smell). I recall there being a predisposition for all execution to happen in `docker-compose up`, and not pre-commands, but I didn't understand that position. I've seen a few "production" docker setups, and running a setup script before isn't uncommon.